src/ulid.cr

```diff
@@ -19,6 +20,53 @@ module ULID
     encode_time(seed_time, TIME_LEN) + encode_random(RANDOM_LEN)
   end
 
+  # Validate a string is a ULID
+  #
+  # ```
+  # ULID.valid?("01B3EAF48P97R8MP9WS6MHDTZ3")
+  # # => true
+  # ```
+  def valid?(ulid : String) : Bool
+    # Incorrect length
+    return false unless ulid.size == TIME_LEN + RANDOM_LEN
+    # Invalid chars
+    return false unless ulid.upcase.chars.all? &.in?(ENCODING)
+
+    return true
+  end
+
+  # Validate a string is a ULID
+  #
+  # ```
+  # ULID.valid!("01B3EAF48P97R8MP9WS6MHDTZ3")
+  # # => nil (or exception)
+  # ```
+  def valid!(ulid : String) : Nil
+    # Incorrect length
+    raise IncorrectLength.new("ULIDs must be 26 characters.") unless ulid.size == TIME_LEN + RANDOM_LEN
+    # Invalid chars
+    raise InvalidChars.new("Invalid characters found.") unless ulid.upcase.chars.all? &.in?(ENCODING)
+
+    return nil
+  end
```

Comment:
I would do this

```suggestion
  # Validate a string is a ULID
  #
  # ```
  # ULID.valid?("01B3EAF48P97R8MP9WS6MHDTZ3")
  # # => true
  # ```
  def valid?(ulid : String) : Bool
    valid(ulid, raise_error: false)
  end

  # Validate a string is a ULID
  #
  # ```
  # ULID.valid!("01B3EAF48P97R8MP9WS6MHDTZ3")
  # # => nil (or exception)
  # ```
  def valid!(ulid : String) : Nil
    valid(ulid, raise_error: true)
    nil
  end

  private def valid(ulid : String, raise_error : Bool) : Bool
    # Incorrect length
    length_error = !(ulid.size == TIME_LEN + RANDOM_LEN)
    raise IncorrectLength.new("ULIDs must be 26 characters.") if raise_error && length_error
  
    # Invalid chars
    char_error = !(ulid.upcase.chars.all? &.in?(ENCODING))
    raise InvalidChars.new("Invalid characters found.") if raise_error && char_error
  
    length_error && char_error
  end
```

Reply:
Or this to save some some computation but looks a bit less elegant
```crystal
private def valid(ulid : String, raise_error : Bool) : Bool?
  # Incorrect length
  unless ulid.size == TIME_LEN + RANDOM_LEN
    raise_error ? raise IncorrectLength.new("ULIDs must be 26 characters.") : return false
  end

  # Invalid chars
  unless ulid.upcase.chars.all? &.in?(ENCODING)
    raise_error ? raise InvalidChars.new("Invalid characters found.") : return false
  end

  true
end
```